src/libstd/io/mod.rs

```diff
@@ -172,14 +172,11 @@ pub trait Read {
     /// Read all bytes until EOF in this source, placing them into `buf`.
     ///
     /// All bytes read from this source will be appended to the specified buffer
-    /// `buf`. This function will return a call to `read` either:
+    /// `buf`. This function will continuously call `read` to append more data to
+    /// `buf` until `read` returns either `Ok(0)` or an error which is not of the
+    /// kind `ErrorKind::Interrupted`.
```

Comment:
I know you're just re-arranging the words here, but an error which _isn't_ that kind? This sentence seems very strange.